addon-manager/impl/src/main/java/org/jboss/forge/addon/manager/impl/ui/DistributionUpdateCommand.java

```diff
@@ -0,0 +1,171 @@
+      {
+         return Results.fail("Updating distribution was aborted.");
+      }
+
+      List<Coordinate> distributionVersions = dependencyResolver.resolveVersions(DependencyQueryBuilder
+               .create(CoordinateBuilder.create()
+                        .setGroupId("org.jboss.forge")
+                        .setArtifactId("forge-distribution")));
+      Coordinate latestDistributionCoordinate;
+      int i = 0;
+      do
+      {
+         i++;
+         if (distributionVersions.size() <= i)
+         {
+            // if we haven't found final version
+            return Results.fail("Have not found any online Final version of the distribution.");
+         }
+         latestDistributionCoordinate = distributionVersions.get(distributionVersions.size() - i);
+      }
+      while (latestDistributionCoordinate.isSnapshot());
+      Version latestVersion = new SingleVersion(latestDistributionCoordinate.getVersion());
+      Version installedVersion = currentAddon.getId().getVersion();
+      if (latestVersion.compareTo(installedVersion) <= 0)
+      {
+         return Results.fail("Forge is already in the most updated version.");
+      }
+      File updateDir = new File(OperatingSystemUtils.getForgeHomeDir().getAbsolutePath() + "/.update");
+
+      Dependency dependency = dependencyResolver.resolveArtifact(DependencyQueryBuilder.create(CoordinateBuilder
+               .create()
+               .setGroupId(latestDistributionCoordinate.getGroupId())
+               .setArtifactId(latestDistributionCoordinate.getArtifactId())
+               .setVersion(latestDistributionCoordinate.getVersion()).setClassifier("offline").setPackaging("zip")));
+      FileResource<?> artifact = dependency.getArtifact();
+      ZipFile zipFile = new ZipFile(artifact.getFullyQualifiedName());
+      File tempFolder = OperatingSystemUtils.createTempDir();
+      String tempFolderPath = tempFolder.getAbsolutePath();
+      zipFile.extractAll(tempFolderPath);
+      FilenameFilter unzippedForgeFilter = new FilenameFilter()
+      {
+         public boolean accept(File dir, String name)
+         {
+            String lowercaseName = name.toLowerCase();
+            if (lowercaseName.contains("forge-distribution"))
+            {
+               return true;
+            }
+            else
+            {
+               return false;
+            }
+         }
+      };
+      File[] forgeDistrubution = new File(tempFolderPath).listFiles(unzippedForgeFilter);
+      String distributionExtractedPath = forgeDistrubution[0].getAbsolutePath();
+      FileUtils.copyDirectory(new File(distributionExtractedPath), updateDir);
+      deleteWholeDirectory(new File(distributionExtractedPath));
+      return Results.success("Sucessfully prepared to be updated to version "
+               + latestDistributionCoordinate.getVersion()
+               + ". Please restart forge.");
+   }
+   
+   private void deleteWholeDirectory(File directory) throws IOException
+   {
```

Comment:
No, this class is in furnace-impl, we cannot use it unless we move it to furnace-api